testsuite/integration/basic/src/test/java/org/wildfly/test/integration/microprofile/health/MicroProfileHealthTestBase.java

```diff
@@ -60,15 +59,37 @@
 @RunAsClient
 public abstract class MicroProfileHealthTestBase {
 
+    public static final String JAR_ARCHIVE_NAME = "MicroProfileHealthTestCaseJar.jar";
+    public static final String WAR_ARCHIVE_NAME = "MicroProfileHealthTestCase.war";
+    public static final String EAR_ARCHIVE_NAME = "MicroProfileHealthTestCase.ear";
+
     abstract void checkGlobalOutcome(ManagementClient managementClient, String operation, boolean mustBeUP, String probeName) throws IOException;
 
-    @Deployment(name = "MicroProfileHealthTestCase", managed = false)
-    public static Archive<?> deploy() {
-        WebArchive war = ShrinkWrap.create(WebArchive.class, "MicroProfileHealthTestCase.war")
-                .addClasses(TestApplication.class, TestApplication.Resource.class, MyProbe.class, MyLiveProbe.class, HealthConfigSource.class)
+    @Deployment(name = EAR_ARCHIVE_NAME, managed = false)
+    public static Archive<?> deployEar() {
+        JavaArchive jarArchive = ShrinkWrap.create(JavaArchive.class, JAR_ARCHIVE_NAME)
+                .addClass(HealthConfigSource.class)
```

Comment:
As mentioned in https://issues.redhat.com/browse/WFLY-12835?focusedCommentId=14115173&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14115173, the config source of a subdeployment will not support reading from another library in the EAR deployment